docker-compose.yml

```diff
@@ -2,34 +2,83 @@ volumes:
   node_modules:
 
 services:
-  regular_redis:
-    image: redis
+  # Redis Master
+  redis-master:
+    image: bitnami/redis:7.4
+    environment:
+      - REDIS_REPLICATION_MODE=master
+      - ALLOW_EMPTY_PASSWORD=yes
     ports:
-      - "36379:6379"
-
-  bitnami_redis:
-    platform: 'linux/amd64'
-    user: root
-    image: 'bitnami/redis:6.2'
+      - "6379:6379"
+  # Redis Replicas
+  redis-replica-1:
+    image: bitnami/redis:7.4
+    environment:
+      - REDIS_REPLICATION_MODE=slave
+      - REDIS_REPLICA_IP=redis-replica-1
+      - REDIS_MASTER_HOST=redis-master
+      - ALLOW_EMPTY_PASSWORD=yes
+    ports:
+      - "6380:6379"
+    depends_on:
+      - redis-master
+  redis-replica-2:
+    image: bitnami/redis:7.4
     environment:
+      - REDIS_REPLICATION_MODE=slave
+      - REDIS_REPLICA_IP=redis-replica-2
+      - REDIS_MASTER_HOST=redis-master
       - ALLOW_EMPTY_PASSWORD=yes
     ports:
-      - "36380:6379"
+      - "6381:6379"
+    depends_on:
+      - redis-master
+
+  # Redis Sentinel nodes
+  redis-sentinel:
+    image: bitnami/redis-sentinel:7.4
+    environment:
+      - REDIS_SENTINEL_QUORUM=1
+      - REDIS_SENTINEL_ANNOUNCE_IP=sentinel
+      - REDIS_MASTER_HOST=redis-master
+      - REDIS_MASTER_SET=redis-master
+    ports:
+      - "26379:26379"
+    depends_on:
+      - redis-master
+      - redis-replica-1
+      - redis-replica-2
 
+  # BullMQ Exporter App
   app:
     image: "node:alpine"
     depends_on:
-      - regular_redis
-      - bitnami_redis
+      - redis-sentinel
     volumes:
       - ".:/app"
       - "node_modules:/app/node_modules"
     working_dir: /app
     environment:
-      - "REDIS_HOST=regular_redis"
-      - "REDIS_PORT=6379"
+      - "EXPORTER_SENTINEL_HOSTS=redis-sentinel:26379"
+      - "EXPORTER_SENTINEL_NAME=redis-master"
+      - "EXPORTER_PREFIX=bull"
+      - "EXPORTER_STAT_PREFIX=bull_queue_"
+      - "EXPORTER_AUTODISCOVER=true"
+      - "EXPORTER_QUEUES=agentic_pdf_to_html_step"
       - "WAIT_UNTIL_READY=TRUE"
+    ports:
+      - "9538:9538"
     command:
-      - tail
-      - -f
-      - /dev/null
+      - sh
+      - -c
+      - |
+        echo "Installing dependencies..."
+        yarn install
+        echo "Building the project..."
+        yarn build
+        echo "Creating test queues..."
+        node test-queues.js
+        echo "Waiting for Sentinel to be fully ready..."
+        sleep 5
+        echo "Starting BullMQ exporter..."
+        node dist/src/index.js --sentinel-hosts "redis-sentinel:26379" --sentinel-name "redis-master" --prefix "bull" --metric-prefix "bull_queue_" --autoDiscover agentic_pdf_to_html_step
```

Comment:
looks like a command for a very specific test, do we want it as default for local app start? mb wrap this (lines 75-84) in `npm run test` instead for ppl to call manually when they want and leave the "blank" `tail -f /dev/null` by default?